cmd/prompt.go

```diff
@@ -0,0 +1,402 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	goprompt "github.com/c-bata/go-prompt"
+	"github.com/c-bata/go-prompt/completer"
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/nsf/termbox-go"
+	"github.com/openconfig/goyang/pkg/yang"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+var promptMode bool
+var promptHistory []string
+var schemaTree *yang.Entry
+var promptModeCmd = &cobra.Command{
+	Use:     "prompt-mode",
+	Aliases: []string{"prompt"},
+	Short:   "enter the interactive gnmic prompt-mode",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := pathCmdRun(promptDirs, promptFiles, promptExcluded, true)
+		if err == nil {
+			promptMode = true
+		}
+		// load history
+		promptHistory = make([]string, 0, 256)
+		home, err := homedir.Dir()
+		if err == nil {
+			content, err := ioutil.ReadFile(home + "/.gnmic.history")
+			if err == nil {
+				history := strings.Split(string(content), "\n")
+				for i := range history {
+					if history[i] != "" {
+						promptHistory = append(promptHistory, history[i])
+					}
+				}
+			}
+		}
+		return err
+	},
+	PostRun: func(cmd *cobra.Command, args []string) {
+		cmd.ResetFlags()
+		initPromptFlags(cmd)
+	},
+	SilenceUsage: true,
+}
+
+var promptQuitCmd = &cobra.Command{
+	Use:   "quit",
+	Short: "quit the gnmic-prompt",
+	Run: func(cmd *cobra.Command, args []string) {
+		// save history
+		home, err := homedir.Dir()
+		if err == nil {
+			f, err := os.Create(home + "/.gnmic.history")
+			if err == nil {
```

Comment:
is it possible to keep the err == nil path with the less indentation?
meaning handle err !=nil first then err == nil

Reply:
Yes, it is possible.